File: order-service/src/main/java/com/nhd/order_service/response/ApiResponse.java
package com.nhd.order_service.response;

import java.time.LocalDateTime;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ApiResponse<T> {
    private LocalDateTime timestamp;
    private int statusCode;
    private String message;
    private T data;

    public ApiResponse(T data, int statusCode, String message) {
        this.timestamp = LocalDateTime.now();
        this.statusCode = statusCode;
        this.data = data;
        this.message = message;
    }
}

File: frontend/src/types/api-response.ts
export interface ApiResponse<T> {
  statusCode: number;
  error: string | null;
  message: string;
  data: T;
}

export type ApiSuccess<T> = ApiResponse<T> & {
  error: null;
};

export type ApiError = ApiResponse<null> & {
  error: string;
};
